pyaerocom/aeroval/coldatatojson_helpers.py

```diff
@@ -426,8 +426,8 @@ def _process_one_station_weekly(stat_name, i, repw_res, meta_glob, time):
 
     ts_data = {
         "time": time,
-        "seasonal": {"obs": yeardict, "mod": yeardict},
-        "yearly": {"obs": yeardict, "mod": yeardict},
+        "seasonal": {"obs": deepcopy(yeardict), "mod": deepcopy(yeardict)},
+        "yearly": {"obs": deepcopy(yeardict), "mod": deepcopy(yeardict)},
```

Comment:
I understand the need for `deepcopy` to create different copies of the same empty dictionary, but would not be better to use `collections.defaultdict` (see [docs])?

Here is how it looks like with `collections.defaultdict`
``` python
from collections import defaultdict
...

    ts_data = {
        "time": time,
        "seasonal": {"obs": defaultdict(dict), "mod": defaultdict(dict)},
        "yearly": {"obs": defaultdict(dict), "mod": defaultdict(dict)},
    }
```

[docs]: https://docs.python.org/3/library/collections.html#collections.defaultdict

Reply:
I just did not know about defaultdict, but did you see the init of `yeardict`?
https://github.com/metno/pyaerocom/blob/d0b18cd996536f5af87ba0c492e86979024fc536/pyaerocom/aeroval/coldatatojson_helpers.py#L517-L520

`yeardict` has some predifined keys which wouldn't be in your `defaultdict(dict)`, right?